lib/receiver/receiver_impl.cc

```diff
@@ -192,7 +192,14 @@ namespace gr
         sch_search_handler(input, noutput_items);
         break;
       case synchronized:
-        synchronized_handler(input, input_items, noutput_items);
+	synchronized_handler(input, input_items, noutput_items,0);
+	size_t inputs_to_process = d_cell_allocation.size();
+	if (d_process_uplink)
+        	inputs_to_process *= 2;
+	for(int input_nr=0;input_nr<inputs_to_process;input_nr++){
+		boost::thread t1(&receiver_impl::synchronized_handler,this,input, input_items, noutput_items, input_nr);
+		usleep(200);
```

Comment:
You should replace this usleep with [join](https://www.boost.org/doc/libs/1_67_0/doc/html/thread/thread_management.html#thread.thread_management.tutorial.join)
You should store all thread objects in a container (array/vector/list) and then in a second loop you should join them one by one.